ext/taglib_base/taglib_base.i

```diff
@@ -34,13 +34,25 @@ namespace TagLib {
 
 // Rename setters to Ruby convention (combining SWIG rename functions
 // does not seem to be possible, thus resort to some magic)
+// We used to do this with one "command" filter, but support for that was
+// removed in SWIG 4.1.0. This is a bit uglier because we need one filter for
+// setOne, one for setOneTwo, and one for setOneTwoThree. Looks like we don't
+// need one for 4 yet.
 // setFoo -> foo=
-%rename("%(command: perl -e \"print lc(join('_', split(/(?=[A-Z])/, substr(@ARGV[0], 3)))), '='\" )s",
-        regexmatch$name="^set[A-Z]") "";
+%rename("%(regex:/set([A-Z][a-z]*)([A-Z][a-z]*)([A-Z][a-z]*)/\\L\\1_\\2_\\3=/)s",
+        regexmatch$name="^set([A-Z][a-z]*){3}$") "";
+%rename("%(regex:/set([A-Z][a-z]*)([A-Z][a-z]*)/\\L\\1_\\2=/)s",
+        regexmatch$name="^set([A-Z][a-z]*){2}$") "";
+%rename("%(regex:/set([A-Z][a-z]*)/\\L\\1=/)s",
+        regexmatch$name="^set([A-Z][a-z]*)$") "";
```

Comment:
@fusion2004 This works (`\L` means make everything after that lowercase), see [Advanced renaming support](https://www.swig.org/Doc4.1/SWIGDocumentation.html#SWIG_advanced_renaming) in SWIG docs:

> The back-references are replaced with the contents of the corresponding capture group while the escape sequences perform the case conversion in the substitution string: `\l` and `\L` convert to the lower case

Reply:
Ah yea, that part I had seen and understood. What I was confused about was trying to get it to prepend underscores to a dynamic number of capital letters, but I see the "uglier" path you took here haha. Definitely simpler!

Reply:
@robinst maybe it would be easier to split this PR in two? First merge the "`%rename` via regex" bit, which should have a tiny diff, and then afterwards "regenerate using SWIG 4.1.1"?

Reply:
Or did you run 6b3fe2e6e4856ba60436bee013f74d69cbbf6a59 through SWIG 4.0.2?

Reply:
I have manually verified that running https://github.com/robinst/taglib-ruby/commit/6b3fe2e6e4856ba60436bee013f74d69cbbf6a59 through SWIG 4.0.2 with TagLib 1.11.1 yields only trivial changes (embedded path comments).

That proves that the new `%rename` directives work as intended.

Reply:
> I have manually verified that running https://github.com/robinst/taglib-ruby/commit/6b3fe2e6e4856ba60436bee013f74d69cbbf6a59 through SWIG 4.0.2 with TagLib 1.11.1 yields only trivial changes (embedded path comments).

Ah that's a great idea, thanks @jacobvosmaer! Cool, so I think we're good to go with this.